### ISSUE TITLE
iOS: Using lineOffset with a geojson shape throws an exception or crashes the app

### ISSUE THREAD

**Issue by hampelm** [nitaliano/react-native-mapbox-gl#1076]
This may affect Android as well but I haven't had a chance to check.  To reproduce: 

Clone the current react-native-mapbox-gl repository (6.0.3 as of today).
Install and run the example app; browse to the GeoJSON Source example to confirm it loads.

Edit `GeoJSONSource.js`:

1. Switch the `smileyFaceFill` layer to a `MapboxGL.LineLayer`
2. Edit the `layerStyles` to include an offset: 

```
const layerStyles = MapboxGL.StyleSheet.create({
  background: {
    backgroundPattern: gridPattern
  },
  smileyFace: {
    lineColor: 'white',
    lineOffset: 5,
    lineWidth: 5
  }
});
```

If you reload the app and browse to the example, the app will crash to the home screen. If you remove the lineOffset, it works fine. It doesn't seem to matter if I pass a FeatureCollection or a Feature. In an app I'm building, I occasionally get this error instead of a hard crash:  

![simulator screen shot - iphone 6 - 2018-02-27 at 13 24 04](https://user-images.githubusercontent.com/86435/36748027-a9b1e4be-1bc4-11e8-8da8-207723621575.png)

**Comment by nitaliano** [comment]
yes, this is a known bug right now it's due to a typecast error on the iOS side, this should not happen on Android

**Comment by AareFabrik** [comment]
Hello,

I have the same problem, is there any progress or a workaround until it is fixed?

**Comment by nitaliano** [comment]
@xiswar you can try to use `lineTranslate` it takes an array `[x, y]` https://github.com/mapbox/react-native-mapbox-gl/blob/master/docs/LineLayer.md#name-7, it might work but that depends on what you're trying to do

**Comment by AareFabrik** [comment]
@nitaliano Thanks for the help. But I would like to create rainbow-like polyLines like here: 

https://maptime-ams.github.io 

This is not possible with lineTranslate as i would have to calculate for every segment of the polyLine the direction and then apply on the direction the right lineTranslate [x, y]. But this creates gabs between the segments of the polyLine. So this is not a possibility for me.

Any other advice?

**Comment by AareFabrik** [comment]
Any progress on this issue?

**Comment by kristfal** [comment]
No progress here, but any PR is very welcome.
